bindings/node.js/Makefile

```diff
@@ -1,40 +1,78 @@
-all: clean build format test bundle
+# For a less verbose yarn.
+YARN = yarn --silent
 
+.PHONY: all
+all: format build test bundle
+
+# Cleans native dependency, bindings and typescript artifacts
+.PHONY: clean
 clean:
-	rm -rf build
-	rm -rf dist
-	rm -f *.node
-	rm -f *.a
-	rm -f *.o
-	rm -rf node_modules
-
-build: src/kzg.cxx lib/kzg.ts package.json binding.gyp Makefile
-	cd ../../src && make c_kzg_4844.o && cp c_kzg_4844.o ../bindings/node.js
-	yarn install
-	yarn node-gyp rebuild
-
-test: build
-	yarn jest
-
-format:
-	yarn prettier --write .
-
-bundle: clean
-	mkdir dist
-	cp README.md dist/README.md
-	cp package.json dist/package.json
-	cp binding.dist.gyp dist/binding.gyp
-	node_modules/.bin/tsc -p tsconfig.build.json
-	cp -r src dist/src
-	mkdir -p dist/deps/c-kzg
-	cp -r ../../blst dist/deps
-	cp ../../src/c_kzg_4844.c dist/deps/c-kzg
-	cp ../../src/c_kzg_4844.h dist/deps/c-kzg
-
-publish: bundle
-	cd dist
-	npm publish
-
-linux-test: bundle
-	docker build -t "linux-test" .
-	docker logs --follow `docker run -d linux-test`
+	@rm -rf build
+	@rm -rf dist
+	@rm -rf deps
+	@rm -f *.node
+	@rm -f *.a
+	@rm -f *.o
+	@rm -rf ref-tests
+
+# Cleans typescript dependencies
+.PHONY: clean-install
+clean-install:
+	@rm -rf node_modules
+
+# Installs typescript dependencies
+.PHONY: install
+install:
+	@$(YARN) install --ignore-scripts
+
+# Cleans and rebuilds native dependencies, bindings and typescript wrapper
+.PHONY: build
+build: install clean
+	@# Prepare the dependencies directory
+	@mkdir -p deps/c-kzg
+	@cp -r ../../blst deps
+	@cp ../../src/c_kzg_4844.c deps/c-kzg
+	@cp ../../src/c_kzg_4844.h deps/c-kzg
+	@# Patch the blst_aux.h to fix a compiler error
+	@awk '{gsub(/typedef struct \{\} blst_uniq;/, "typedef struct { int dummy; } blst_uniq;")}1' \
+		deps/blst/bindings/blst_aux.h > blst_aux_temp.h
+	@mv blst_aux_temp.h deps/blst/bindings/blst_aux.h
+	@# Build the bindings
+	@$(YARN) node-gyp --loglevel=warn configure
+	@$(YARN) node-gyp --loglevel=warn build
+	@$(YARN) tsc -p tsconfig.build.json
+
+# Bundle the distribution, also helpful for cross compatibility checks
+.PHONY: bundle
+bundle: build
+	@mv deps dist
```

Comment:
Would it be ok to `cp` here instead of `mv` so that subsequent runs of `make test` when updating the test file don't require `build` to be run again?

Reply:
So `build` is phony (meaning it's not backed by a file & will run every time) and `build` will run `clean` first anyway, so those files won't exist. But a `cp` would work here too. It should have the same result.